src/chain/chain.ts

```diff
@@ -30,15 +30,18 @@ export class BeaconChain extends EventEmitter {
   private db: BeaconDB;
   private eth1: IEth1Notifier;
   private _latestBlock: BeaconBlock;
+  private logger: ILogger;
 
-  public constructor(opts, {db, eth1}) {
+  public constructor(opts, {db, eth1, logger}: {db: BeaconDB; eth1: IEth1Notifier; logger: ILogger}) {
```

Comment:
I think we very recently added a `IBeaconDB` interface.
Can you use that instead of `BeaconDB` in all of these cases throughout? I think you'll need to also change thinkgs like the line above (L30) to `private db: IBeaconDB;`

Reply:
Updated with the latest commit. But according to Marin's suggestion will this implementation is going to merged ? Or we should use the previous implementation as pass context as optional parameter in logger?